src/Document/Obstacles/TextureMatrix.ts

```diff
@@ -0,0 +1,37 @@
+import { Vector4F } from '../../Utilities/types';
+import { INameable } from '../Attributes/INameable';
+import {
+  bzwFloat,
+  bzwFloatVector,
+  bzwString,
+  bzwVector4F,
+} from '../attributeParsers';
+import { IBaseObject, newIBaseObject } from './BaseObject';
+
+export const TextureMatrixProperties = {
+  name: bzwString,
+  scale: bzwVector4F,
+  spin: bzwFloat,
+  shift: bzwFloatVector,
+  center: bzwFloatVector,
+  fixedscale: bzwFloatVector,
+  fixedspin: bzwFloat,
+  fixedshift: bzwFloatVector,
+};
+
+export interface ITextureMatrix extends IBaseObject, INameable {
+  scale?: Vector4F;
+  spin?: number;
+  shift?: number[];
+  center?: number[];
+  fixedscale?: number[];
+  fixedspin?: number;
+  fixedshift?: number[];
```

Comment:
```suggestion
  shift?: [number, number];
  center?: [number, number];
  fixedscale?: [number, number];
  fixedspin?: number;
  fixedshift?: [number, number];
```

I believe these are U/V coordinates according to the [Wiki](https://wiki.bzflag.org/TextureMatrix) and Man page. Would there be a time that there'd be more than just two numbers for these?

Reply:
I don't think there would ever be more than two numbers.